src/caffe/solver.cpp

```diff
@@ -188,10 +205,20 @@ void Solver<Dtype>::Step(int iters) {
     }
 
     if (param_.test_interval() && iter_ % param_.test_interval() == 0
-        && (iter_ > 0 || param_.test_initialization())) {
+        && (iter_ > 0 || param_.test_initialization())
+        && Caffe::root_solver()) {
       TestAll();
     }
 
+    timer.Start();
+    timing.str("");
+    timing << "Timing ";
+    if (param().solver_mode() == SolverParameter_SolverMode_GPU) {
+      timing << "(device " << param().device_id() << ") ";
+    }
+    for (int i = 0; i < callbacks_.size(); ++i) {
+      callbacks_[i]->on_start(&timer, &timing);
+    }
     const bool display = param_.display() && iter_ % param_.display() == 0;
```

Comment:
must add 'timer.Start();' here to restart timer, or line 266 that timing for grads maybe incorrect.